core/object/worker_thread_pool.cpp

```diff
@@ -165,10 +165,20 @@ void WorkerThreadPool::_native_low_priority_thread_function(void *p_user) {
 }
 
 void WorkerThreadPool::_post_task(Task *p_task, bool p_high_priority) {
+	if (threads.size() <= 1) {
```

Comment:
If the pool has one thread, this would keep it unused. To me, this check should be `== 0`. However, I think that such a thing can't happen:
https://github.com/godotengine/godot/pull/72704/files#diff-63ffe8a2a76ddbe86885b63c8d9e96cc2bd7ac6180e1e2fac118fc76a53f04cdR413-R417

Reply:
That would be ideal, although as you've seen yourself, `WorkerThreadPool` seems to be treating `threads.size() == 0` as uninitialized, and I was afraid of breaking any hidden assumptions there.

`<= 1` was a quick compromise to make it possible to get this behavior right now, while also generalizing to `threads.size() == 0` if that ever becomes allowed.

Reply:
Today:
- `threads.size() == 0` -> Wrong, uninitialized.
- `threads.size() >= 1` -> Use >= 1 threads.
With your PR:
- `threads.size() == 0` -> Wrong, uninitialized. But if not considered uninitialized anymore, would be run-on-caller.
- `threads.size() == 1` -> Assume zero threads, so not even the only one we have is used. **This is where I see the problem.**
- `threads.size() >= 2` -> Use >= 1 threads.